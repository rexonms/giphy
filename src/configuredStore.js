import { createStore, compose, applyMiddleware } from 'redux';
import createHistory from 'history/createBrowserHistory';
import { connectRouter, routerMiddleware } from 'connected-react-router';
import thunk from 'redux-thunk';

import createReducer from './reducers';

export const history = createHistory();

function configuredStore(initialState = undefined) {
  const composeEnhancers = process.env.NODE_ENV !== 'production ' ? window.__REDUX_DEVTOOLS_EXTENSION_COMPOSE__ || compose : compose;
  const store = createStore(
    connectRouter(history)(createReducer()),
    initialState,
    composeEnhancers(applyMiddleware(
      routerMiddleware(history),
      thunk,
    )),
  );
  store.asyncReducers = {};
  return store;
}

export default configuredStore;
