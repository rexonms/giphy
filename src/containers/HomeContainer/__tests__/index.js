import React from 'react';
import { shallow } from 'enzyme';

import ConnectedHomeContainer, { HomeContainer } from './../index';
import configuredStore from './../../../configuredStore';

describe('HomeContainer container', () => {
  let props;
  let shallowHomeContainer;
  const homeContainer = () => {
    if (!shallowHomeContainer) {
      shallowHomeContainer = shallow(<HomeContainer {...props} />);
    }
    return shallowHomeContainer;
  };
  beforeEach(() => {
    props = {
      home: {},
      getGifs: jest.fn(),
    };
    shallowHomeContainer = undefined;
  });
  it('should match the previous snapshot', () => {
    expect(homeContainer()).toMatchSnapshot();
  });
  it('should trigger componentWillMount', () => {
    jest.spyOn(HomeContainer.prototype, 'componentWillMount');
    homeContainer();
    expect(HomeContainer.prototype.componentWillMount.mock.calls.length).toBe(1);
  });
});

describe('Connected HomeContainer container', () => {
  const props = {
    welcome: {},
    getGifs: jest.fn(),
  };
  const wrapper = shallow(<ConnectedHomeContainer {...props} store={configuredStore()} />).dive();
  expect(wrapper).toMatchSnapshot();
});
