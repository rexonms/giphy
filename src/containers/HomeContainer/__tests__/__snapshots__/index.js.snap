// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[` 1`] = `<LoadingBars />`;

exports[`HomeContainer container should match the previous snapshot 1`] = `<LoadingBars />`;
